src/js/reducers/pipelines.js

```diff
@@ -0,0 +1,74 @@
+/* eslint new-cap:0 */
+'use strict';
+
+var Immutable = require('immutable');
+
+var actions = require('../constants/actions');
+var counter = require('../util/counter');
+var immutableUtils = require('../util/immutable-utils');
+var get = immutableUtils.get;
+var getIn = immutableUtils.getIn;
+var set = immutableUtils.set;
+var setIn = immutableUtils.setIn;
+var ensureValuePresent = immutableUtils.ensureValuePresent;
+var ensureValueAbsent = immutableUtils.ensureValueAbsent;
+var assign = require('object-assign');
+
+
+/**
+ * Main pipelines reducer function, which generates a new state for the
+ * pipelines (pipelines) property store based on the changes specified by the
+ * dispatched action object.
+ *
+ * @param {Object} state - An Immutable.Map state object
+ * @param {Object} action - A redux action object
+ * @returns {Object} A new Immutable.Map with the changes specified by the action
+ */
+function pipelinesReducer(state, action) {
+  if (typeof state === 'undefined') {
+    return Immutable.fromJS({
+      pipelines: []
+    });
+  }
+
+  var pipelines = state.get("pipelines")
+
+  if (action.type === actions.CREATE_PIPELINE) {
+    var newPipelines = pipelines.push(Immutable.fromJS({
+      name: action.id,
+      id: counter.type("pipeline"),
+      source: null
+    }));
```

Comment:
This looks good to me for the most part. One thing to keep in mind with objects in our store: the keys/properties of the object should correspond to things found in the output Vega specification. Any Lyra-specific things should be prefixed with an underscore. The `export` methods then automatically strip these Lyra-specific properties out. 

It matters less with Pipelines since we never export them directly (they're just a helper structure to group together related Datasets). But, for consistency, we should use `_id` instead of `id`. `source` is probably fine as is. 

If in doubt, follow the naming conventions found in the existing class definitions or feel free to ping me via Slack :)